### ISSUE TITLE
Provide Feedback on Rholang Tutorial

### ISSUE THREAD

**Issue by TrenchFloat** [rchain/Members#919]
Let's give helpful, detailed feedback on @JoshOrndorff's [Rholang Tutorial](https://github.com/JoshOrndorff/LearnRholangByExample).

NOTE: Joshy's tutorial and this issue both target those **who do NOT know rholang!** Whether you have never written code before or simply haven't learned this language yet, the goal of this tutorial is to get you writing in rholang. If it doesn't, your feedback is _valuable!_

A new and improved version of the rholang tutorial will be released at the start of RCon3 (Sept 5) or a few days before. Joshy and I are looking for 6 or more reviewers of the Rholang Tutorial to work through the 12 lessons and write detailed feedback on what could be more clear or on anything that isn't straightforward. The key word here is _detailed!_ Feedback should include visual appeal, word choice, suggested intermediate steps, any undue assumptions made on the reader's coding knowledge, or pitfalls experienced along the way.

Once the tutorial is released, Reviewers will start GitHub repositories for working through the examples of the lessons. Written reviews can be posted to the repositories, and are most helpful if done in chunks. A reviewer can give feedback on all 12 lessons right off the bat, and continue to participate by responding to updates on the tutorial, or he/she can review a few lessons at a time. Review of the tutorial should continue until the launch of mainnet or until Joshy is totally satisfied with the product, whichever comes first.

Joshy is holds one office hour each week at [https://zoom.us/j/6853551826](https://zoom.us/j/6853551826) to give support to anyone working through the tutorial (whether reviewing or not) and hear their thoughts. Rewards will be based on the more formal written reviews, though these weekly conversations would be to a reviewer's benefit.

If you are interested in reviewing some or all of the anticipated rholang tutorial, please assign yourself to this issue! No commitment necessary.

### Budget and Objective

Reviewers contributing fully will receive a proposed reward of $150 each month, for a (proposed) potential of $600 for the four months until the launch of mainnet. Reviewers can also write on some but not all lessons for less than the proposed reward. Rewards scale with amount & quality of feedback.

_Estimated Budget of Task:_ **$3600**
_Estimated Timeline Required to Complete the Task:_ **Sep. 5 to Launch of mainnet**
_How will we measure completion?_ **The tutorial's works effectively to Joshy's satisfaction**

**Comment by allancto** [comment]
@TrenchFloat i strongly support this. From what I've seen of Joshy's work it's high quality and I believe everyone who participates will learn and will become more fluent with the technology. 

Thanks for posting this. 
-@allancto

**Comment by TrenchFloat** [comment]
I'm glad to see interest in this issue! Just to confirm, @Ojimadu, @Viraculous, and @David405, none of you know yet how to code in rholang?

**Comment by mjreitz** [comment]
I'm interested in helping as well! I don't have any experience in coding so I can help with making it as user-friendly as possible.

**Comment by Ojimadu** [comment]
@TrenchFloat I could write some basic smart contracts at this point.

**Comment by ogini1** [comment]
I'm interested in this issue, and I'll like to provide some feedback but I can't assign myself.

**Comment by David405** [comment]
Hi @ogini1 I presume you are new to RChain. You are not yet a collaborator on this repository and thus may not be able to assign yourself to an issue, send to mail to ops@rchain.coop to request a collaborator's role and also register on www.rchain.coop to become a member of RChain.

**Comment by David405** [comment]
@TrenchFloat my proficiency level with rholang at this point is intermediate.

**Comment by TrenchFloat** [comment]
@ojimadu @David405 The target participants for this issue are those with no experience with rholang - but your feedback is just as valuable! But sure to denote your rholang experience somewhere in your reviews, and please help us find 6 non-rholang-developers! With myself, mjreitz, ogini1, and Viraculous (assuming ogini1 and Viraculous are not rholang coders) we have 4 - of course, the more reviewers, the better.

**Comment by ogini1** [comment]
Thanks @David405 I appreciate the help I'll do that now

**Comment by Keaycee** [comment]
@ogini1 you can follow this steps below.

## To contribute to RChain, please do (if you haven’t done so already):
 
- [ ] Watch video https://www.youtube.com/watch?v=vcIZSZmpO9E&feature=youtu.be
- [ ] Watch presentation: https://docs.google.com/presentation/d/1B2Vu8o3ACwruY6HY1ayXRQ4qkNKsMy4hdbOdxrCHI2o/edit#slide=id.g3827e46a00_0_68
- [ ] Join rchain@googlegroups.com to get access to documents of RChain https://groups.google.com/forum/#!forum/rchain/join
- [ ] Join github/rchain/bounties in order to contribute https://github.com/rchain/bounties/invitations
- [ ] Fill out the RAM registration form https://docs.google.com/forms/d/e/1FAIpQLSecwGUVFNx_Xa_Qsw5bxLnaKstPS8kQnfrUGqpuf22rLDteDg/viewform 
- [ ] Collaborate in all kind of issues https://github.com/rchain/bounties/issues
- [ ] Collaborate in development https://rchain.atlassian.net/browse/OPS-31?jql=labels%20%3D%20bounty
- [ ] Fool around with our Bounty App https://rewards.rchain.coop/
- [ ] Read about collaborating https://github.com/rchain/bounties/blob/master/CONTRIBUTING.md
- [ ] Read about our decentralized bounty system. https://github.com/rchain/bounties/wiki/How-To-Use-the-Budget-Rewards-Web-App
- [ ] Join RChain Coop by signing up for membership http://member.rchain.coop/
- [ ] Claim your reward: https://docs.google.com/forms/d/e/1FAIpQLSe3ZxbwVL_yQZ7DTSw9V5VZod1U9XeZxDMZB7jupRCnx79erQ/viewform
- [ ] fill out a W8 https://www.irs.gov/pub/irs-pdf/fw8ben.pdf
- [ ] read W8 instructions https://docs.google.com/document/d/1LdxfZEjM37ub_qFizM_otrp2X1YxiJh7VDpfqeYDfu8/edit?usp=sharing
- [ ] or fill out a W9 https://www.irs.gov/pub/irs-pdf/fw9.pdf
- [ ] W9 instructions https://docs.google.com/document/d/1hg83LDU__6VYV_fkZTeZrErCPKs8PsHGasRVAvQG_uY/edit

**Comment by Keaycee** [comment]
I'm interested to review all of the rholang tutorials. Although i have no knowledge on rholang. But i'm open to learn.

**Comment by Keaycee** [comment]
Can we get started through
![screenshot 128](https://user-images.githubusercontent.com/34557673/44628326-b33cb700-a93d-11e8-8640-5484e8f4606e.png)

**Comment by Tonyprisca13** [comment]
Am interested in learning @TrenchFloat

**Comment by TrenchFloat** [comment]
@Keaycee Joshy is releasing a new version of the tutorial very soon. Of course, you can let Joshy know your thoughts anytime, but he could have already changed the subject of your feedback, so feedback will be most helpful after the new release.

**Comment by allancto** [comment]
All, might this issue be slightly expanded to include a review of the invoice system #912? The code and process designed to be simple and auditable by non-programmers. One of the important design goals is to be transparent and easy to audit. The style of code is intended to be as simple as possible and have few external dependencies. The code isn't yet complete, but our target is to have it ready for use in parallel with our existing declarations system for August (so, live by Sept 9). 

The plan is to have it run as a web app on a dedicated server staged and hosted by the Colab sysadmin channel. The code isn't yet ready for review but this [Usage and Code Overview](https://docs.google.com/document/d/1-lPFVxnfdFEwz56cfJVm6CpbwYDPYC0CUAESeYdxULA/) doc should give you a good idea of what it will be like. We'll try also to get a simple demo of what it might be like to use up in the next day or two. 

Comments welcome, directly in the doc. Thanks! -@allancto

**Comment by Viraculous** [comment]
Is there any tutorial around the installation of RNode 0.6.1 on Ubuntu?

**Comment by David405** [comment]
@viraculous are you aware 0.6.2 is the most recent version? Or you intentionally want to install 0.6.1?

**Comment by Viraculous** [comment]
Nope,  thanks for the info. I will have to install that one to see how it works

**Comment by David405** [comment]
If you have any issues installing or running rnode, you can message me on discord to provide you a walkthrough

**Comment by Viraculous** [comment]
Ok

**Comment by pmoorman** [comment]
Maybe helpful to other people also, here's the repo (so you can create a PR with changes): https://github.com/joshorndorff/LearnRholangByExample

**Comment by TrenchFloat** [comment]
@JoshOrndorff and I will be around (IOU a zoom link) at 7:00 AM Seattle time on Thursdays to gather and collect feedback/help out with the tutorial. Of course, feedback is welcome anytime!

**Comment by TrenchFloat** [comment]
This week we'll have a loose topic of discussion - lesson 0-3 of the [Rholang Tutorial](https://github.com/JoshOrndorff/LearnRholangByExample). Please come on Thursday with feedback, questions, and ideas for these lessons. Of course, discussion is not restricted to these topics!

**Comment by Viraculous** [comment]
@JoshOrndorff is there a prerequisite file to be installed before running the "rnode eval intersection.rho"? I getting an error note #File not found.

**Comment by JoshOrndorff** [comment]
To be most helpful, I'll need to see exactly what command you typed and exactly what output you got.

Before you can use `rnode eval` you need to have a running node (standalone is fine). I've given some details in lesson 0 about setting up the development environment.

https://github.com/JoshOrndorff/LearnRholangByExample/blob/master/0-DeveloperEnvironment/README.md#local-node

If you're really stuck, use rchain.cloud for now and come to office hours Thursday morning at 7:00AM Pacific time.

**Comment by JoshOrndorff** [comment]
I'd like to revive this issue. @TrenchFloat Can you continue to help me manage this issue?

**Comment by TrenchFloat** [comment]
Absolutely, @JoshOrndorff - sorry to let this issue get a little dusty. I've been taking some informal notes on Joshy's Thursday office hours (except 10/4, when I was absent), but it seems those blocks of time have veered into topics not tutorial-centered. We'll have to focus more on independent reviewing and feedback-giving, it looks like.

Joshy, could you clear up what form you'd like reviews to be in, in order to be rewarded? You mentioned keeping independent GitHub repos for the tutorial. Is @pmoorman's comment reward-able?

**Comment by TrenchFloat** [comment]
I'll throw out a proposal for rewards for this month. Discussion welcome (as always).

@pmoorman  made it through 80% of the tutorial. 150*.8= $120 for Pieter this month.
$100 to myself and @JoshOrndorff for start-up and organization - expected reward to be much less for me the next 3 months, and more for Joshy, who will be implementing all the feedback.

I haven't seen any tutorial-specific feedback in Joshy's Thursday hours, so I'm considering that separate from this issue. I'm open ears to examples of tutorial discussion on 10/4 which I missed.

**Comment by JoshOrndorff** [comment]
@TrenchFloat The budgets you've asked for you and @pmoorman are low enough that I'm totally comfortable with them. I'm paid through SOW so I won't need any bounty payment. I've voted $220 (100 for you and 120 for Pieter as you've suggested.)

I never bring an agenda to my office hours. Rather we talk about whatever the attendees bring. So if you want to talk about the tutorial, please bring it up. I'm totally fine with that. I'm also fine with talking about other rholang related ideas.

Pieter's comments are quite valuable, but moving forward I will not accept comments written here. Each reviewer should fork my original github repository (https://github.com/joshorndorff/LearnRholangByExample) complete the examples, and provide corrections and suggestions as pull requests. If anyone is struggling to do that or doesn't know github well enough to do it, please ask during colab or hit me up directly.

I will not accept "I did the work but didn't make the PR, here's a link to some other format".

**Comment by ogini1** [comment]
@JoshOrndoff Please I would like to know if it is enough to use rchain.cloud for the tutorial since that will be most convenient for me at this point.

**Comment by ogini1** [comment]
@JoshOrndorff I would like to know if it is enough to use rchain.cloud for the tutorial since that would be most convenient at this time.

**Comment by ogini1** [comment]
For me, that is.

**Comment by JoshOrndorff** [comment]
rchain.cloud is pretty much sufficient. Just remember that you will not have your own personal tuplespace to play with there.

**Comment by Barkov-F** [comment]
In tutorial on developer environment (lesson 0) REPL stands for (run execute print loop). I thought it should be "read" instead of "run". Can I also suggest to add a link to the[ ItelliJ idea rholang plugin](https://plugins.jetbrains.com/plugin/9833-rholang)? Great tutorial, I will keep on reading it attentively.

**Comment by rolandkofler** [comment]
@JoshOrndorff it's not on github pages, is it published somewhere else or in maintainance?
https://joshorndorff.github.io/LearnRholangByExample/lesson1-SendingAndStandardOut/index.html

**Comment by JoshOrndorff** [comment]
@Barkov-F I've fixed the REPL acronym. Thanks for pointing that out. Is the IntelliJ plugin a complete way to _run_ rholang or is it syntax highlighting?

@rolandkofler You're correct, it is not on github pages. The best palce to read it right now is directly from the README files. It is also kind of hosted at rchain.coop/learn-rholang, but sadly it is not kept up to date there.

**Comment by Barkov-F** [comment]
@JoshOrndorff The plugin is for syntax highlighting, right. So I guess it doesnt belong to runtime env, but the section is called "Developer Environment" and I thought it would be ok to include it. So I did in RUS variant of the tutorial. 
One more question: is it terminologically correct to call `!`, `!!` and such "operators" or there is another term for actions performed on channels?

**Comment by Barkov-F** [comment]
In [Receiving](https://github.com/JoshOrndorff/LearnRholangByExample/tree/master/2-Receiving) lesson an exercise for coffeeShop.rho says "Order a second drink" but in code there are already two drinks. May be change that for "a third drink" or just "more drinks"?

**Comment by JoshOrndorff** [comment]
@Barkov-F I'm not 100% sure whether it is "correct" to call `!` the send operator, but it sounds fine to my ear. I can't think of anything more correct to call it.

"order another drink" issue, you can fork and make a PR. Then you'll be earning some of the available bounty and saving me manual work.

Anyone else who wants bounty can also make those PRs.

**Comment by JoshOrndorff** [comment]
Congrat to @Barkov-F who was the first to express interest in this bounty _and_ create a PR https://github.com/JoshOrndorff/LearnRholangByExample/pull/20
